scipy/sparse/_index.py

```diff
@@ -32,74 +26,124 @@ class IndexMixin:
     """
     This class provides common dispatching and validation logic for indexing.
     """
-    def _raise_on_1d_array_slice(self):
-        """We do not currently support 1D sparse arrays.
-
-        This function is called each time that a 1D array would
-        result, raising an error instead.
-
-        Once 1D sparse arrays are implemented, it should be removed.
-        """
-        from scipy.sparse import sparray
-
-        if isinstance(self, sparray):
-            raise NotImplementedError(
-                'We have not yet implemented 1D sparse slices; '
-                'please index using explicit indices, e.g. `x[:, [0]]`'
-            )
-
     def __getitem__(self, key):
-        row, col = self._validate_indices(key)
+        index, new_shape = self._validate_indices(key)
+
+        # 1D array
+        if len(index) == 1:
+            idx = index[0]
+            if isinstance(idx, np.ndarray):
+                if idx.shape == ():
+                    idx = idx.item()
+            if isinstance(idx, INT_TYPES):
+                res = self._get_int(idx)
+            elif isinstance(idx, slice):
+                res = self._get_slice(idx)
+            else:  # assume array idx
+                res = self._get_array(idx)
+
+            # package the result and return
+            if not isinstance(self, sparray):
+                return res
+            if res.shape == () and new_shape != ():
+                if len(new_shape) == 1:
+                    return self.__class__([res], shape=new_shape, dtype=self.dtype)
+                if len(new_shape) == 2:
+                    return self.__class__([[res]], shape=new_shape, dtype=self.dtype)
```

Comment:
This is probably fine, but I don't 100% get which case we're handling here.

Reply:
I've added a comment indicating that this is to handle multiple `np.newaxis` indices, i.e. `A[3, 4, None, None]`.